src/Futhark/CLI/Bench.hs

```diff
@@ -297,7 +298,22 @@ runBenchmarkCase opts futhark program entry pad_to tr@(TestRun _ input_spec (Suc
         putStr $ descString dataset_desc pad_to
 
       reportResult runtimes
-      return $ Just $ DataResult dataset_desc $ Right (runtimes, errout)
+      ( case getMemoryUsage errout of
+          Just bytes ->
+            Right (runtimes, bytes, errout)
+          Nothing ->
+            Left "Could not parse memory usage from profiling output"
+        )
+        & DataResult dataset_desc
+        & Just
+        & return
+
+getMemoryUsage :: T.Text -> Maybe Int
+getMemoryUsage t =
+  let mem_regex = "Peak memory usage for space '[^']+': ([0-9]+) bytes." :: T.Text
```

Comment:
What happens if there are multiple memory spaces?

Reply:
I am not really sure what this means, but cf. https://github.com/diku-dk/futhark/pull/1215/files/b4cb1dc490436b353326ca8268d1cde447a7961c#diff-c8ca25d5bf1f5928bbd214a58075cdf189a2dbbdee9007c06fceb8225d4238b0R315-R316, we will return `Nothing`. What would you prefer to happen?

Reply:
This should return a mapping from memory space to the peak, and the JSON should contain a dict.

We will certainly use both CPU and GPU memory in the same program in the relatively near future, and we might do even more funky things in the distant future, who knows.